### DISCUSSION TITLE
Use HLS4ML produced model without Jupyter Notebook?

### DISCUSSION THREAD

**Discussion by abigailbutkaUF** [fastmachinelearning/hls4ml#1028] [Q&A]
Hi all!

I am working on a project that needs a NN model in the hardware that is fed data **directly from the PL side, not from the ZYNQ processor.** I have the full HLS4ML flow working, and can run the Jupyter Notebook tests, but have been unable to determine an easy way to move to the hardware.

I was thinking I could replicate the AXI transactions sent to the DMA from the PYNQ, but the transactions I am able to capture using the ILA do not correlate to any of my known test data. I have also tried looking at the predict() code to see if there were any hints there, but I am not the most familiar with PYNQ overlay.

Is what I am trying to do possible? Am I looking in the wrong places here? If anyone has any suggestions or documentation to recommend I would greatly appreciate it.

Thank you! :)

**Comment by jmitrevs** [comment]
If you use the Vitis (2022.2 or later) or Vivado (2020.1 or earlier) backends, the output is HLS for an IP that you can integrate by yourself in a Vivado project. Alternately, you can also write your own HLS wrapper for the produced HLS code.
